bigquery/cloud-client/user_credentials_test.py

```diff
@@ -0,0 +1,62 @@
+# Copyright 2017 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+import google.auth
+import google_auth_oauthlib.flow
+from user_credentials import auth_query
+
+
+PROJECT = os.environ['GCLOUD_PROJECT']
+
+
+class MockFlow(object):
+    def __init__(self, credentials):
+        self.credentials = credentials
+
+    @classmethod
+    def from_client_secrets_file(cls, client_secrets, scopes=None):
+        credentials, _ = google.auth.default(scopes=scopes)
+        return cls(credentials)
+
+    def run_local_server(self):
+        pass
+
+    def run_console(self):
+        pass
+
+
+def test_auth_query_console(capsys):
+    # Arrange
+    google_auth_oauthlib.flow.InstalledAppFlow = MockFlow
```

Comment:
Better yet, I'd actually prefer you to write a new fixture for this using Mock's `autospec`. This will ensure that even if google-auth-oauthlib's API changes, our test will catch the interface change *despite* the mocking:

```python
@pytest.fixture
def mock_flow():
    flow_patch = mock.patch('google_oauthlib_flow.InstalledAppFlow', autospec=True)

    with flow_patch as flow_mock:
        flow_mock.credentials = google.auth.default()[0]
        yield flow_mock
```